pipeline/steam_extract.py

```diff
@@ -0,0 +1,171 @@
+                'span', class_='user_reviews_count').text.strip('()'))
+
+        else:
+            return 0.00
+        if negative:
+            n_num = int(negative.find(
+                'span', class_='user_reviews_count').text.strip('()'))
+        else:
+            return 100
+
+        return round((p_num/(p_num+n_num)) * 100, 2)
+
+
+def get_platforms(game_soup: BeautifulSoup) -> list:
+    """Function to get the platforms a game is available on from its URL"""
+
+    if game_soup.findAll(
+            'div', class_="sysreq_tab"):
+        platforms = [i.text.strip() for i in game_soup.findAll(
+            'div', class_="sysreq_tab")]
+        for i in range(len(platforms)):
+            if "Linux" in platforms[i]:
+                platforms[i] = "Linux"
+        return platforms
+    else:
+        platforms = game_soup.findAll(
+            'div', class_="sysreq_contents")
+        platform_list = [platform.find(
+            'ul', class_='bb_ul').text for platform in platforms if platform.find(
+                'ul', class_='bb_ul')][0]
+        if "Windows" in platform_list:
+            return "Windows"
+        if "mac" in platform_list:
+            return "macOS"
+        if "Linux" in platform_list:
+            return "Linux"
+
+        return "Windows"
+
+
+def get_tags(game_soup: BeautifulSoup) -> list:
+    """Function to get the tags of a game from its URL"""
+    return [i.text.strip()
+            for i in game_soup.find_all('a', class_='app_tag')]
+
+
+def get_developer(game_soup: BeautifulSoup) -> str:
+    """Function to get the developing company of a game from its URL"""
+    return game_soup.find('div', class_='dev_row').text.split('\n')[-2]
+
+
+def get_publisher(game_soup: BeautifulSoup) -> str:
+    """Function to get the publisher of a game from its URL"""
+    pub = game_soup.find_all('div', class_='dev_row')
+    pub_list = [x.text.split('\n') for x in pub]
+    for each in pub_list:
+        if 'Publisher:' in each:
+            return each[-2]
+    return None
+
+
+def get_name_price_date(container: BeautifulSoup) -> list:
+    """Function that returns a list containing the:
+      title, price and date of a game from its from the search page"""
+    title = container.find('span', {'class': 'title'}).text
+    price = container.find('div', {'class': "discount_final_price"}).text
+    if price == "Free":
+        float_price = 0.00
+    else:
+        float_price = float(price[1:])
+    release_date = container.find(
+        'div', {'class': "col search_released responsive_secondrow"}).text
+
+    return [title, float_price, release_date.strip()]
+
+
+def get_description(game_soup: BeautifulSoup) -> str:
+    """Function to get the description of a game from its URL"""
+    return game_soup.find('div', id='game_area_description').text.strip()
+
+
+def get_platform_ids(platform: list) -> list:
+    '''Returns a list of platform IDs given a
+    string of playable platforms.'''
```

Comment:
Something I suggested to Shindy earlier today - for assigning platform names to their ids, we could use a global `dict` object assigning each platform name to its id?

So something like this:

```
{'Windows': 1, 'macOS': 2, 'Linux': 3}
```